app/elements/app-element.html

```diff
@@ -202,26 +99,27 @@ <h2 class="paper-font-display2">You are not authorized to see this page.</h2>
 			},
 			logout: function() {
 				localStorage.clear();
-				this.refreshMenu(false);
+				this.refreshMenu();
 				page.redirect('/login');
 			},
-			refreshMenu: function(loggedIn) {
-				this.isCompany = !!util.organization();
-				this.isCoder = !this.isCompany;
-				this.isLoggedIn = loggedIn;
+			refreshMenu: function() {
+				this.$.appNav.refresh();
 			},
 			displayInstalledToast: function() {
 				console.log('Caching complete.');
 			},
-			createChallenge: function() {
-				this.$.taskList.createChallenge();
-			},
 			inviteByToken: function(token){
-				this.$.invitationRequest.url = util.build('/invite/auth/' + token);
-				this.$.invitationRequest.generateRequest();
-			},
-			onInvitationResponse: function(r){
-				page('/challenge/' + r.detail.detail.response);
+				var invitationRequest = document.createElement('iron-ajax');
+				invitationRequest.withCredentials = true;
+				invitationRequest.url = util.build('/invite/auth/' + token);
+				invitationRequest.onResponse = this.onInvitationResponse;
+				var self = this;
+				invitationRequest.addEventListener('response', function(r){
+					localStorage.user = JSON.stringify({id: 0, name: 'mocked'});
```

Comment:
the old logic was `this.isCoder = !this.isCompany` wich is soo wrong because if the user isn`t login he`ll appear as a coder. And our check now is `!!util.user()` wich will be removed after i remove all localStorage usages.